com.microsoft.mrtk.spatialmanipulation/Solvers/TapToPlace.cs

```diff
@@ -467,66 +494,93 @@ protected virtual void SetRotation()
+            {
+                Debug.Log($"Failed to register the pick up event: {e}");
+            }
+        }
+
+        /// <summary>
+        /// Registers the event which performs placement.
+        /// </summary>
+        private void RegisterPlacementEvent()
+        {
+            try
+            {
+                interactable.OnClicked.AddListener(StopPlacement);
+            }
+            catch (Exception e)
+            {
+                Debug.Log($"Failed to register the placemenent event: {e}");
             }
-            interactable.OnClicked.AddListener(StartPlacement);
         }
 
         /// <summary>
         /// Unregisters the input action which performs placement.
         /// </summary>
         private void UnregisterPlacementAction()
         {
-            InputAction placementAction = GetInputActionFromReference(placementActionReference);
-            if (placementAction == null)
+            foreach (InputAction placementAction in lastActions)
             {
-                Debug.Log("Failed to unregister the placement action, the action reference was null or contained no action.");
-                return;
+                if (placementAction == null)
+                {
+                    Debug.Log("Failed to unregister the placement action, the action reference was null or contained no action.");
+                    continue;
+                }
+                placementAction.performed -= StopPlacementAction;
             }
-            placementAction.performed -= StopPlacement;
+            lastActions = new List<InputAction>();
         }
 
         /// <summary>
         /// Unregisters the event which performs pickup.
         /// </summary>
-        private void UnregisterPickupAction()
+        private void UnregisterPickupEvent()
         {
-            if (interactable == null)
+            try
             {
-                Debug.Log("Failed to unregister the pick up event. There is no StatefulInteractable set.");
-                return;
+                interactable.OnClicked.RemoveListener(StartPlacement);
+            }
+            catch (Exception e)
+            {
+                Debug.Log($"Failed to unregister the pick up event: {e}");
             }
-            interactable.OnClicked.RemoveListener(StartPlacement);
         }
 
         /// <summary>
-        /// Extracts the InputAction from the InputActionReference.
+        /// Unregisters the event which performs placement.
         /// </summary>
-        /// <param name="actionReference">
-        /// The InputActionReference containing the desired InputAction.
-        /// </param>
-        /// <returns>An InputAction, or null.</returns>
-        public static InputAction GetInputActionFromReference(InputActionReference actionReference)
+        private void UnregisterPlacementEvent()
         {
-            if (actionReference == null) { return null; }
-            return actionReference.action;
+            try
```

Comment:
Remove try/catch

interactable since is required.  If it's not there we want Unity to throw an exception.  It's expectational and we should prevent code from running any further.

Reply:
Same comment for everywhere else you added a try/catch